src/main/java/uk/ac/ebi/eva/pipeline/io/VepProcess.java

```diff
@@ -0,0 +1,341 @@
+        int exitValue = process.exitValue();
+        if (exitValue != 0) {
+            String timestamp = Long.toString(System.currentTimeMillis());
+            String backupVepOutput = annotationParameters.getVepOutput().replaceFirst("tsv\\.gz$",
+                    timestamp + ".tsv.gz");
+
+            String errorLog = backupVepOutput + ".errors.txt";
+            try {
+                connectStreams(process.getErrorStream(), new FileOutputStream(errorLog));
+            } catch (IOException e) {
+                throw new ItemStreamException("VEP exited with code " + exitValue
+                        + " but the file to dump the errors could not be created: " + errorLog,
+                        e);
+            }
+            boolean renamed = new File(annotationParameters.getVepOutput()).renameTo(new File(backupVepOutput));
+            if (renamed) {
+                logger.info("Renamed VEP output to " + backupVepOutput + " as backup because VEP failed");
+            }
+            throw new ItemStreamException("Error while running VEP (exit status " + exitValue + "). See "
+                    + errorLog + " for the errors description from VEP.");
+        }
+    }
+
+    private void checkOutputWritingStatus() {
+        try {
+            outputCapturer.join(timeoutInSeconds * CONVERT_SECONDS_TO_MILLISECONDS);
+        } catch (InterruptedException e) {
+            throw new ItemStreamException("Interrupted while waiting for the VEP output writer thread to finish. ", e);
+        }
+        if (outputCapturer.isAlive()) {
+            outputCapturer.interrupt();
+            throw new ItemStreamException("Reached the timeout (" + timeoutInSeconds
+                    + " seconds) while waiting for VEP output writing to finish. Killed the thread.");
+        }
+        if (!writingOk.get()) {
+            throw new ItemStreamException("VEP output writer thread could not finish properly. ");
+        }
+    }
+
+    /**
+     * read all the vep output in inputStream and write it into the outputStream, logging the coordinates once in each
+     * chunk.
+     *
+     * @param reader must be closed externally
+     * @param writer must be closed externally
+     * @param skipComments if false, will write all lines starting with '#', if true, will not write any.
+     * @return written lines.
+     */
+    private long copyVepOutput(BufferedReader reader, OutputStreamWriter writer, boolean skipComments)
+            throws IOException {
+        long writtenLines = 0;
+
+        String line = getNextLine(reader, skipComments);
+        String lastLine = line;
+        while (line != null) {
+            writer.write(line);
+            writer.write('\n');
+            writtenLines++;
+
+            lastLine = line;
+            line = getNextLine(reader, skipComments);
+        }
+
+        writer.flush();
+        outputIdleSince.set(System.currentTimeMillis());
+        logCoordinates(lastLine, writtenLines);
```

Comment:
It won't, because VEP fails if it gets no input, and we take care of that in VepAnnotationFileWriter. The only way I can imagine that, is VEP getting a variant and writing nothing. but well, I can add a check for that:

    logger.warn("VEP didn't write anything, this might be a symptom of a previous error");